cpp/oneapi/dal/algo/pca/backend/gpu/finalize_train_kernel_cov_dpc.cpp

```diff
@@ -50,38 +139,51 @@ auto compute_eigenvectors_on_host(sycl::queue& q,
 
     auto eigvecs = pr::ndarray<Float, 2>::empty({ component_count, column_count });
     auto eigvals = pr::ndarray<Float, 1>::empty(component_count);
-
     auto host_corr = corr.to_host(q, deps);
     pr::sym_eigvals_descending(host_corr, component_count, eigvecs, eigvals);
 
     return std::make_tuple(eigvecs, eigvals);
 }
 
 template <typename Float>
-auto compute_correlation(sycl::queue& q,
-                         std::int64_t row_count,
-                         const pr::ndview<Float, 2>& xtx,
-                         const pr::ndarray<Float, 1>& sums,
-                         const bk::event_vector& deps = {}) {
-    ONEDAL_PROFILER_TASK(compute_correlation, q);
-    ONEDAL_ASSERT(sums.has_data());
-    ONEDAL_ASSERT(xtx.has_data());
-    ONEDAL_ASSERT(xtx.get_dimension(1) > 0);
-
-    const std::int64_t column_count = xtx.get_dimension(1);
-
-    auto tmp = pr::ndarray<Float, 1>::empty(q, { column_count }, alloc::device);
+auto compute_singular_values_on_host(sycl::queue& q,
+                                     pr::ndarray<Float, 1> eigenvalues,
+                                     std::int64_t row_count,
+                                     const dal::backend::event_vector& deps = {}) {
+    const std::int64_t component_count = eigenvalues.get_dimension(0);
 
-    auto corr = pr::ndarray<Float, 2>::empty(q, { column_count, column_count }, alloc::device);
+    auto singular_values = pr::ndarray<Float, 1>::empty(component_count);
 
-    auto copy_event = copy(q, corr, xtx, { deps });
+    auto eigvals_ptr = eigenvalues.get_data();
+    auto singular_values_ptr = singular_values.get_mutable_data();
+    for (std::int64_t i = 0; i < component_count; i++) {
```

Comment:
I insist on using same style everywhere. In this case we typically use pre-increment instead

```suggestion
    for (std::int64_t i = 0; i < component_count; ++i) {
```